cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -1080,13 +1099,26 @@ instance Structured NoIndexCache
 
 -- | We need to save only .cabal file contents
 instance Binary NoIndexCacheEntry where
-    put (CacheGPD _ bs) = put bs
+    put (CacheGPD _ bs) = do
+      put (0 :: Word8)
+      put bs
+    put (NoIndexCachePreference dep) = do
+      put (1 :: Word8)
+      put dep
 
     get = do
-        bs <- get
-        case parseGenericPackageDescriptionMaybe bs of
-            Just gpd -> return (CacheGPD gpd bs)
-            Nothing  -> fail "Failed to parse GPD"
+        -- TODO: fendor, read it out of the cache, too
```

Comment:
TODO

Reply:
thanks, actually addressed that todo, even. Just forgot to remove it